### PR TITLE
Enhance conceal - fix uc(first) concealing and enable customization of conceal marker

### PR DESCRIPTION
**Pull request by reaysawa**
This is the current master branch

image

### This branch

image

- Fixes %uc(first) regex conceal
- Adds g:neosnippet#conceal_char for customizing your marker, defaults to '|'

### CODE REVIEW

File: autoload/neosnippet.vim [hunk 1]
```diff
@@ -28,6 +28,8 @@ call neosnippet#util#set_default(
       \ 'g:neosnippet#enable_auto_clear_markers', 1)
 call neosnippet#util#set_default(
       \ 'g:neosnippet#enable_complete_done', 0)
+call neosnippet#util#set_default(
+      \ 'g:neosnippet#conceal_char', '|')
```
Comment (Shougo) [inline autoload/neosnippet.vim]: Please add the documentation for `g:neosnippet#conceal_char`.

File: autoload/neosnippet/init.vim [hunk 2]
```diff
@@ -57,14 +57,16 @@ function! s:initialize_others() abort
...
-            \ syntax region neosnippetConcealExpandSnippets
-            \ matchgroup=neosnippetExpandSnippets
-            \ start='<`\d\+:\=\%(#:\|TARGET:\?\)\?\|<{\d\+:\=\%(#:\|TARGET:\?\)\?\|<|'
-            \ end='`>\|}>\||>'
-            \ containedin=ALL
-            \ cchar=|
-            \ concealends oneline
+      let start = '<`0\|<`\|<{\d\+:\=\%(#:\|TARGET:\?\)\?\|%\w\+(<|'
+      let end = '\(:\w\+\|:#:\w\+\)\?`>\|}>\||>)\?'
+      execute " autocmd BufNewFile,BufRead,Syntax * " .
```
Comment (Shougo) [inline autoload/neosnippet/init.vim]: Please use single quotes instead of double quotes.

File: autoload/neosnippet/init.vim [hunk 3]
```diff
@@ -57,14 +57,16 @@ function! s:initialize_others() abort
...
+            \ " syntax region neosnippetConcealExpandSnippets " .
+            \ " matchgroup=neosnippetExpandSnippets " .
+            \ " start='" . start . "' " .
```
Comment (Shougo) [inline autoload/neosnippet/init.vim]: I think using `printf()` is better for it.

```vim
printf(' start=%s end=%s', string(start), string(end))
```

**Comment by Shougo** [comment]
Please fix my suggested problems.

**Comment by reaysawa** [comment]
@Shougo I have made the changes you requested. Do you want to use this branch a bit to see if the regex is broken somewhere? Then I could try to fix before merging.

**Comment by Shougo** [comment]
OK.  I will try it later.

**Comment by Shougo** [comment]
No problem.  Merged.

**Pull request merged**